### DISCUSSION TITLE
HA doesn't connect to 33KTL-A

### DISCUSSION THREAD

**Discussion by JenDa666** [wlcrs/huawei_solar#655] [Connecting to the inverter]
My setup:
Huawei SUN2000 33ktl-A
Huawei Smartlogger 3000
Huawei DTSU666-H
I know the huawei_solar doesn't work with the SM3000. Now I try to connect the HA dirctly with the Huawei 33ktl-A but it dosen't work.

The cable from SM is connected to RS485_1 OUT (2+3) and HA to RS485_2 OUT (6 + 8):
![ppIMG_20240317_094230~2](https://github.com/wlcrs/huawei_solar/assets/160950936/d66ddc3f-db1f-45fc-86a4-80f731bea0d6)
![IMG_20240317_110826](https://github.com/wlcrs/huawei_solar/assets/160950936/ba09c841-5434-40fd-be86-17e0cea3e2a4)

The settingon on the inverter and HA should be ok:
![Modbus not connected](https://github.com/wlcrs/huawei_solar/assets/160950936/395c498e-9437-4cc7-8b39-362a4a2c60a2)
![Screenshot_2024-03-17-11-03-42-842_com huawei smartpvms](https://github.com/wlcrs/huawei_solar/assets/160950936/5cf0dac8-6395-49cc-8b29-4c405e8059ed)

Where is the mistake?
THX in advance!

**Comment by didoisp** [comment]
Were you able to connect?

**Reply by JenDa666** [reply]
no, I use a external shelly

**Comment by nurzen** [comment]
Have you tried a low level connection https://github.com/wlcrs/huawei-solar-lib
You need to change the example for a serial connection.

**Reply by JenDa666** [reply]
No, the modbus adresses of the 33ktl-A are different. Thats the problem.

**Reply by didoisp** [reply]
Can you tell me more about that ?  If it is only modbus adresses diferent it is not problem

**Reply by JenDa666** [reply]
Here are examples for the different addresses and the documentations: 
![Modbus Protocol](https://github.com/user-attachments/assets/c8eba650-5de4-4956-b8fb-e58f4a8e3389)
https://www-file.huawei.com/~/media/CORPORATE/PDF/FusionSolar/HUAWEI_SUN2000_245KTL28KTL_MODBUS_Interface_Definitions_20150715_JP.pdf
https://support.huawei.com/enterprise/de/doc/EDOC1100387581?currentPartNo=k004&togo=content&section=k004

In my understanding you have to fork the project and "just" exchange the one address with the other. I'm not abel to do it by my self. My programming skils are very poor. Please let my know if its works.
